pkg/sql/sqlstats/insights/provider_test.go

```diff
@@ -1,30 +0,0 @@
-// Copyright 2022 The Cockroach Authors.
-//
-// Use of this software is governed by the Business Source License
-// included in the file licenses/BSL.txt.
-//
-// As of the Change Date specified in that file, in accordance with
-// the Business Source License, use of this software will be governed
-// by the Apache License, Version 2.0, included in the file
-// licenses/APL.txt.
-
-package insights
-
-import (
-	"testing"
-
-	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
-	"github.com/cockroachdb/cockroach/pkg/sql/clusterunique"
-	"github.com/stretchr/testify/require"
-)
-
-func TestProvider_WriterIgnoresInternalExecutorObservations(t *testing.T) {
```

Comment:
This test doesn't seem like it was particularly useful for testing the real behaviour 🤔
I filed https://github.com/cockroachdb/cockroach/issues/128220 as a follow up for the obs team.

Reply:
Ya, it was a simple test for the `nullWriter` that didn't test much, so I removed it when I removed the `nullWriter`. I agree that an integration test would be great.